lib/elinux_build_target.dart

```diff
@@ -351,29 +351,34 @@ class NativeBundle {
 
     // Run the native build.
     final String cmakeBuildType = buildMode.isPrecompiled ? 'Release' : 'Debug';
-    final String targetArch = buildInfo.targetArch;
+    final String targetArch = buildInfo.targetArch == 'arm64' ? 'aarch64' : 'x86_64';
     final String hostArch = _getCurrentHostPlatformArchName();
     final String targetCompilerTriple = buildInfo.targetCompilerTriple;
+    final String targetCompilerFlags = buildInfo.targetCompilerFlags;
     final String targetSysroot = buildInfo.targetSysroot;
     final String systemIncludeDirectories = buildInfo.systemIncludeDirectories;
-    final bool needCrossBuild = targetArch != hostArch;
-    final bool needCrossBuildOptionsForArm64 = targetArch == 'arm64';
     RunResult result = await _processUtils.run(
       <String>[
         'cmake',
         '-DCMAKE_BUILD_TYPE=$cmakeBuildType',
         '-DFLUTTER_TARGET_BACKEND_TYPE=${buildInfo.targetBackendType}',
-        if (needCrossBuild) '-DFLUTTER_TARGET_PLATFORM_SYSROOT=$targetSysroot',
-        if (needCrossBuild && systemIncludeDirectories != null)
+        if (targetSysroot != '/') '-DCMAKE_SYSROOT=$targetSysroot',
+        if (buildInfo.targetArch != hostArch) '-DCMAKE_SYSTEM_PROCESSOR=$targetArch',
+        if (systemIncludeDirectories != null)
           '-DFLUTTER_SYSTEM_INCLUDE_DIRECTORIES=$systemIncludeDirectories',
-        if (needCrossBuildOptionsForArm64 && targetCompilerTriple != null)
+        if (targetCompilerTriple != null)
           '-DCMAKE_C_COMPILER_TARGET=$targetCompilerTriple',
-        if (needCrossBuildOptionsForArm64 && targetCompilerTriple != null)
+        if (targetCompilerTriple != null)
           '-DCMAKE_CXX_COMPILER_TARGET=$targetCompilerTriple',
+        if (targetCompilerFlags != null)
+          '-DCMAKE_C_FLAGS=$targetCompilerFlags',
+        if (targetCompilerFlags != null)
+          '-DCMAKE_CXX_FLAGS=$targetCompilerFlags',
+        '-DCMAKE_C_COMPILER=clang',
+        '-DCMAKE_CXX_COMPILER=clang++',
         eLinuxDir.path,
       ],
       workingDirectory: outputDir.path,
-      environment: <String, String>{'CC': 'clang', 'CXX': 'clang++'},
```

Comment:
It looks like you are adding another content, could you please separate the PRs?

Reply:
OK, the extra one is removed, I will open new PR